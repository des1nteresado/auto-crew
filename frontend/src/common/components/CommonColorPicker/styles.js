const styles = (theme) => ({
  root: {
    height: 40,
    width: 80,
    borderRadius: 5,
    backgroundColor: theme.palette.secondary.main,
    color: theme.palette.black,
    cursor: 'pointer',
    textAlign: 'center',
    textTransform: 'uppercase',
    fontWeight: 'bold',
    '&:hover': {
      backgroundColor: `${theme.palette.black} !important`,
      color: theme.palette.white,
    },
    '& .MuiTypography-body2': {
      fontWeight: 'bold',
    },
    '& span': {
      marginRight: '0 !important',
      marginBottom: '0 !important',
    },
  },
  colorSecondary: {
    '&.Mui-checked': {
      backgroundColor: theme.palette.black,
      color: theme.palette.white,
    },
    '&.Mui-disabled': {
      color: theme.palette.text.primary,
      backgroundColor: theme.palette.secondary.main,
    },
  },
});

export default styles;
