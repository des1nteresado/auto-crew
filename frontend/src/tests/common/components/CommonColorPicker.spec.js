import React from 'react';
import { shallow, mount } from 'enzyme';

import CommonColorPicker from 'common/components/CommonColorPicker';

const defaultProps = {
  value: 'value',
  selectedValue: 'value',
};

const setUp = (props) => shallow(<CommonColorPicker {...defaultProps} {...props} />);

describe('CommonColorPicker', () => {
  let component;

  beforeEach(() => {
    component = setUp();
  });

  it('renders CommonColorPicker component correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('sets checked prop to true when selectedValue equals value', () => {
    const colorPicker = mount(<CommonColorPicker {...defaultProps} />);
    const input = colorPicker.find('input');

    expect(input.props().checked).toBeTruthy();
  });
});
