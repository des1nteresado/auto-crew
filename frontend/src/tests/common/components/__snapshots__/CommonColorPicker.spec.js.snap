// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommonColorPicker renders CommonColorPicker component correctly 1`] = `
<CommonColorPicker
  classes={
    Object {
      "colorSecondary": "CommonColorPicker-colorSecondary-2",
      "root": "CommonColorPicker-root-1",
    }
  }
  disabled={false}
  selectedValue="value"
  value="value"
/>
`;
